src/Model/RecordLocale.php

```diff
@@ -209,7 +209,8 @@ public function getLink()
         }
 
         // Get link from localised record
-        return $record->Link();
+        $action = Controller::curr()->getRequest()->params()['Action'] ?? null;
```

Comment:
The problem here is that you are making a model action dependant on controller state, which creates a risky and inverse dependency between model / controller.

Better solution: Allow getLink to take an action (like nearly every other link / relative link method) and conditionally join it.

E.g. 

```
public function getLink($action = null) {
}
```

and remove Controller::curr() in it's entirety. :)

Then you can use `$Link('show')` in your custom template.

Reply:
You could also support this with `$record->LocaleLink($this->getLocale(), $action)`

Reply:
The main reason I have a problem with this code is that Controller::curr() may not necessarily be the controller for the record in question. E.g. Controller::curr() being `Security`, but $record being a `HomePage` class.

It's risky to make this presumption.

Reply:
Better solution:

- add $action to getLink() method
- add getCurrentLink() method, that assumes the current controller action, and calls getLink() with the current action / id / etc...